### PR TITLE
Add OnEnable and OnDisable to NitroxEntity

### PR DESCRIPTION
**Pull request by killzoms**
This theoretically should help with some pointless log spam

### CODE REVIEW

File: NitroxClient/MonoBehaviours/NitroxEntity.cs [hunk 1]
```diff
@@ -115,6 +115,22 @@ public void Start()
             }
         }
 
+        public void OnDisable() // Called with Destroy as well
+        {
+            if (Id != null) // Theoretically shouldnt happen but Unity sucks
+            {
+                gameObjectsById.Remove(Id);
+            }
```
Comment (dartasen) [inline NitroxClient/MonoBehaviours/NitroxEntity.cs:123]: Maybe we should log this happy case if it shouldn't happen ?

File: NitroxClient/MonoBehaviours/NitroxEntity.cs [hunk 2]
```diff
@@ -115,6 +115,22 @@ public void Start()
...
+        }
+
+        public void OnEnable()
+        {
+            if (Id != null)
+            {
+                gameObjectsById[Id] = gameObject;
+            }
```
Comment (dartasen) [inline NitroxClient/MonoBehaviours/NitroxEntity.cs:131]: same here

**Review by dartasen — CHANGES_REQUESTED** [review]

**Comment by Measurity** [comment]
Does this have to do with serialization that skips this ID? If we know what's causing it then it should have a comment in code.

**Comment by Measurity** [comment]
TODO: Verify that this won't prematurely unassign the Nitrox ID (i.e. object is still needed somewhere by our code) when said object is far away.

**Comment by Measurity** [comment]
Can we move this code to OnDestroy instead?

**Comment by killzoms** [comment]
@Measurity done

File: NitroxClient/MonoBehaviours/NitroxEntity.cs [hunk 3]
```diff
@@ -115,6 +115,28 @@ public void Start()
...
+        {
+            if (Id == null)
+            {
+                Log.Error($"Nitrox Id was null on {gameObject.name} in OnEnable");
+                return;
+            }
+
+            gameObjectsById[Id] = gameObject;
+        }
+
```
Comment (Jannify) [inline NitroxClient/MonoBehaviours/NitroxEntity.cs:139]:
```suggestion
        public void OnEnable()
        {
            if (Id == null)
            {
                Log.Error($"Nitrox Id was null on {gameObject.name} in OnEnable");
                return;
            }

            gameObjectsById[Id] = gameObject;
        }

        public void OnDestroy()
        {
            if (Id == null)
            {
                Log.Error($"Nitrox Id was null on {gameObject.name} in OnDestroy");
                return;
            }

            gameObjectsById.Remove(Id);
        }
```

**Review by Jannify — APPROVED** [review]

**Pull request merged**